### PR TITLE
[JDBC] Allow using session's user and password for connecting database in JDBC engine

### PR DESCRIPTION
**Pull request by bowenliang123**
### _Why are the changes needed?_

- allow using the session's user/password to connect database in the JDBC engine
- it is allowed to be applied on any share level of engines, since every kyuubi session maintains a dedicated JDBC connection in the current implementation

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/master/contributing/code/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by bowenliang123** [comment]
cc @zhaomin1423 @pan3793

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 1]
```diff
@@ -2657,6 +2657,13 @@ object KyuubiConf {
       .stringConf
       .createOptional
 
+  val ENGINE_JDBC_USE_SESSION_USER_PASSWORD: ConfigEntry[Boolean] =
+    buildConf("kyuubi.engine.jdbc.connection.use.session.user.password")
```
Comment (pan3793) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:2661]: I recommend to

1. rename it to `kyuubi.engine.jdbc.propagateCredential`
2. make it only takes effect on CONNECTION/USER share level
3. mark as internal

Comment (bowenliang123) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala]: I have limited this feature effecting on CONNECTION/USER share level.

Meanwhile,
In the current implementation of JDBC engine, every session has a dedicated JDBC connection. Therefore, I think we should allow them to custom session's user/password in runtime, even in GROUP and SERVER level which only controls the share level of engine instance but not the exact JDBC connection.

Comment (pan3793) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala]: I wrongly thought that sessions share a connection pool in the JDBC engine, then things become intuitive.
Suggestions 2. 3. could be ignored. Sorry for making you back and forth.

Comment (bowenliang123) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala]: No problem.  You raised a combination of share levels on engines and sessions which is worth to be discussed.

**Review by pan3793 — APPROVED** [review]

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala (hunk 1 above)
Comment (bowenliang123) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala]: Updated the description with the clarification above.

**Comment by bowenliang123** [comment]
Thanks, merged to master.

**Pull request closed without merging**